pandas/tests/reshape/test_merge.py

```diff
@@ -724,6 +724,63 @@ def test_indicator(self):
                           how='outer', indicator=True)
         assert_frame_equal(test5, hand_coded_result)
 
+    def test_validation(self):
```

Comment:
rather than repeating tests I think pull these out to another test file, ``test_api`` and run thru both merge and merge_asof (parametrize is best)